includes/class-wc-stripe-account.php

```diff
@@ -0,0 +1,160 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC_Stripe_Account class.
+ *
+ * Communicates with Stripe API.
+ */
+class WC_Stripe_Account {
+
+	const ACCOUNT_OPTION = 'wcstripe_account_data';
+	const ACCOUNT_API    = 'account';
+
+	public function __construct() {
+		self::cache_account();
+	}
+
+	/**
+	 * Gets and caches the data for the account connected to this site.
+	 *
+	 * @return array Account data or empty if failed to retrieve account data.
+	 */
+	public static function get_cached_account_data() {
+		if ( ! woocommerce_gateway_stripe()->connect->is_connected() ) {
+			return [];
+		}
+
+		$account = self::read_account_from_cache();
+
+		if ( ! empty( $account ) ) {
+			return $account;
+		}
+
+		$account = self::cache_account();
+		return $account;
+	}
+
+	/**
+	 * Read the account from the WP option we cache it in.
+	 *
+	 * @return array
+	 */
+	private static function read_account_from_cache() {
+		$account_cache = get_option( self::ACCOUNT_OPTION );
+
+		if ( false === $account_cache || ! isset( $account_cache['account'] ) || ! isset( $account_cache['expires'] ) ) {
+			// No option found or the data isn't in the shape we expect.
+			return [];
+		}
+
+		// Set $account to empty if the cache has expired, triggering another fetch.
+		if ( $account_cache['expires'] < time() ) {
+			return [];
+		}
+
+		// We have fresh account data in the cache, so return it.
+		return $account_cache['account'];
+	}
+
+	/**
+	 * Caches account data for a period of time.
+	 */
+	private static function cache_account() {
```

Comment:
When the merchant switches "test mode", I'm guessing that this cache will need to be invalidated. Is there something that does it?
Maybe the "test mode" can be part of the cache key.

Reply:
Updated the code. If the current mode does not match the cached mode, fetching the data again.